pkg/config/config.go

```diff
@@ -1633,38 +1671,42 @@ func (c *Config) PublicFields() (map[string]string, error) {
 	return result, nil
 }
 
-func LoadConfig(configFile string) (cfg *Config, err error) {
+func LoadConfig(configFile string) (*Config, error) {
 	var filesToCheck []string
 	if configFile != "" {
 		filesToCheck = append(filesToCheck, configFile)
 	}
 
 	filesToCheck = append(filesToCheck, defaultConfigFiles...)
-	cfg = NewConfig()
+	cfg := NewConfig()
 	cfgMetadata, err := config_loader.LoadYamlConfig(cfg, filesToCheck...)
 	if err != nil {
 		err = fmt.Errorf("%w, %s: %s", ErrUnableToParseConfigFile, configFile, err.Error())
-		return
-	}
-
-	cfg, err = ApplyDatabind(cfg)
-	if err != nil {
-		return
+		return cfg, err
 	}
 
-	templateConfig := NewConfig()
-	templateConfigMetadata, err := config_loader.LoadYamlConfig(templateConfig, filesToCheck...)
-
-	if err != nil {
-		err = fmt.Errorf("%w, %s: %s", ErrUnableToParseConfigFile, configFile, err.Error())
+	if !cfg.Databind.IsEmpty() {
+		dynamicConfig := DynamicConfig{
+			databindSources:        nil,
+			templateConfigMetadata: nil,
+			templateConfig:         nil,
+			refreshedConfig:        nil,
+		}
+		dynamicConfig.databindSources, err = cfg.Databind.DataSources()
```

Comment:
I would move this `cfg.Databind.DataSources()` to the first thing after the if as an early return so later the dynamicConfig struct can be initalized and not use the "setters"

```go
if !cfg.Databind.IsEmpty() {

		dbs, err := cfg.Databind.DataSources()
		if err != nil {
			return cfg, err
		}
		templateConfig := NewConfig()
		_, err = config_loader.LoadYamlConfig(templateConfig, filesToCheck...)
		if err != nil {
			return cfg, err
		}
		
		dynamicConfig := DynamicConfig{
			databindSources:        dbs,
			templateConfigMetadata: cfgMetadata,
			templateConfig:         templateConfig,
			refreshedConfig:        nil,
		}
		
		cfg.dynamicConfig = &dynamicConfig
		cfg, err = applyDatabind(&dynamicConfig)

		if err != nil {
			return cfg, err
		}
	}
```